Cargo.toml

```diff
@@ -56,74 +57,89 @@ primitives = { path = "primitives", default-features = false }
+sp-blockchain = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git" }
+sp-consensus = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-consensus-pow = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-core = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-inherents = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-io = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-keystore = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-offchain = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-runtime = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-runtime-interface = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-session = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-state-machine = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-staking = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-std = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-timestamp = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git" }
+sp-transaction-pool = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-version = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+substrate-build-script-utils = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git" }
+substrate-frame-rpc-system = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git" }
+substrate-prometheus-endpoint = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git" }
+substrate-test-client = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git" }
+tracing = { package = "sp-tracing", branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+frame-try-runtime = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+sp-externalities = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git", default-features = false }
+substrate-wasm-builder = "10.0.0"
```

Comment:
@nathanwhit can you elaborate more on this change. What does it achieve and why is it needed ? 

I'm not sure how this will play out with the upgrade to polkadot-v0.9.42 and switching to a stable rustc version.